src/ui/public/courier/fetch/request/request.js

```diff
@@ -15,12 +15,42 @@ export default function AbstractReqProvider(Private, Promise) {
       this.source = source;
       this.defer = defer || Promise.defer();
       this._whenAbortedHandlers = [];
-
       requestQueue.push(this);
     }
 
+    /**
+     *  Called by the loopers to find requests that should be sent to the
+     *  fetch() module. When a module is sent to fetch() it's _fetchRequested flag
+     *  is set, and this consults that flag so requests are not send to fetch()
+     *  multiple times.
+     *
+     *  @return {Boolean}
+     */
     canStart() {
-      return Boolean(!this.stopped && !this.source._fetchDisabled);
+      return !this._fetchRequested && !this.stopped && !this.source._fetchDisabled;
+    }
+
+    /**
+     *  Used to find requests that were previously sent to the fetch() module but
+     *  have not been started yet, so they can be started.
+     *
+     *  @return {Boolean}
+     */
+    isFetchRequestedAndPending() {
+      return !!this._fetchRequested && !this.started;
```

Comment:
Forces boolean `true` out of anything. I agree that it's unusual since it's part of a larger boolean statement and now I wonder if there's more to the trick than I remember or just a force of habit.